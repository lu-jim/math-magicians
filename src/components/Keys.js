import React from 'react';
import Element from './Element';
import Operator from './Operator';

class Keys extends React.Component {
  constructor(props) {
    super(props);
    this.state = {};
  }

  render() {
    return (
      <div className="keys">
        <Element />
        <Operator />
      </div>
    );
  }
}

export default Keys;
