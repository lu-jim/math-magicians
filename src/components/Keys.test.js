import React from 'react';
import renderer from 'react-test-renderer';
import Keys from './Keys';

test('renders keys correctly', () => {
  const onClick = () => 'mock';
  const tree = renderer.create(<Keys onClick={onClick} />).toJSON();
  expect(tree).toMatchSnapshot();
});
