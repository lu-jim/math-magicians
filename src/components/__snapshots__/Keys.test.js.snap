// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders keys correctly 1`] = `
<div
  className="keys"
>
  <button
    className="element zero"
    onClick={[Function]}
    type="button"
    value="0"
  >
    0
  </button>
  <button
    className="element one"
    onClick={[Function]}
    type="button"
    value="1"
  >
    1
  </button>
  <button
    className="element two"
    onClick={[Function]}
    type="button"
    value="2"
  >
    2
  </button>
  <button
    className="element three"
    onClick={[Function]}
    type="button"
    value="3"
  >
    3
  </button>
  <button
    className="element four"
    onClick={[Function]}
    type="button"
    value="4"
  >
    4
  </button>
  <button
    className="element five"
    onClick={[Function]}
    type="button"
    value="5"
  >
    5
  </button>
  <button
    className="element six"
    onClick={[Function]}
    type="button"
    value="6"
  >
    6
  </button>
  <button
    className="element seven"
    onClick={[Function]}
    type="button"
    value="7"
  >
    7
  </button>
  <button
    className="element eight"
    onClick={[Function]}
    type="button"
    value="8"
  >
    8
  </button>
  <button
    className="element nine"
    onClick={[Function]}
    type="button"
    value="9"
  >
    9
  </button>
  <button
    className="element clear"
    onClick={[Function]}
    type="button"
    value="AC"
  >
    AC
  </button>
  <button
    className="element sign"
    onClick={[Function]}
    type="button"
    value="+/-"
  >
    +/-
  </button>
  <button
    className="element percent"
    onClick={[Function]}
    type="button"
    value="%"
  >
    %
  </button>
  <button
    className="element dot"
    onClick={[Function]}
    type="button"
    value="."
  >
    .
  </button>
  <button
    className="operator plus"
    onClick={[Function]}
    type="button"
    value="+"
  >
    +
  </button>
  <button
    className="operator minus"
    onClick={[Function]}
    type="button"
    value="-"
  >
    -
  </button>
  <button
    className="operator times"
    onClick={[Function]}
    type="button"
    value="x"
  >
    x
  </button>
  <button
    className="operator divide"
    onClick={[Function]}
    type="button"
    value="รท"
  >
    รท
  </button>
  <button
    className="operator equal"
    onClick={[Function]}
    type="button"
    value="="
  >
    =
  </button>
</div>
`;
